spring-boot-project/spring-boot-actuator-autoconfigure/src/test/java/org/springframework/boot/actuate/autoconfigure/metrics/cassandra/CassandraMetricsAutoConfigurationTests.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2012-2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.autoconfigure.metrics.cassandra;
+
+import java.util.Optional;
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.datastax.oss.driver.api.core.CqlSession;
+import com.datastax.oss.driver.api.core.metrics.Metrics;
+import io.micrometer.core.instrument.MeterRegistry;
+import java.time.Duration;
+import org.junit.jupiter.api.Test;
+
+import org.springframework.boot.actuate.autoconfigure.metrics.test.MetricsRun;
+import org.springframework.boot.actuate.metrics.cassandra.CassandraMetricsBinder;
+import org.springframework.boot.autoconfigure.AutoConfigurations;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.context.ApplicationContext;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+/**
+ * Tests for {@link CassandraMetricsAutoConfiguration}.
+ *
+ * @author Erik Merkle
+ */
+public class CassandraMetricsAutoConfigurationTests {
+
+	private ApplicationContextRunner contextRunner = new ApplicationContextRunner().with(MetricsRun.simple())
+			.withConfiguration(AutoConfigurations.of(CassandraMetricsAutoConfiguration.class));
+
+	/**
+	 * Cassandra driver metrics should be enabled by default as long as the desired
+	 * metrics are enabled in the Driver's configuration.
+	 */
+	@Test
+	void autoConfiguredCassandraIsInstrumented() {
+		this.contextRunner.withBean(CqlSession.class, () -> mockSession(true)).run((context) -> {
+			MeterRegistry registry = context.getBean(MeterRegistry.class);
+			// Assert a typical Driver Meter metric
```

Comment:
nit: can we extract assertions of metrics to separate methods for a readability? i.e:
https://github.com/datastax/spring-boot/pull/2/files#diff-8991f02ba760e2c1590560601a62f5a1R59-R61
to `assertMeterMetric()`, then `assertGaugeMetric()`, etc